### PR TITLE
login UI finished

### PR DESCRIPTION
**Pull request by ferran98campos**

### CODE REVIEW

File: next-app/pages/api/auth/index.ts [hunk 1]
```diff
@@ -19,15 +19,16 @@ class AuthHandler extends RequestHandler {
...
             //Check data
-            const parsedBody = JSON.parse(request.body);
-            UserValidation.validateAuthUser(parsedBody as IUser);
+            UserValidation.validateAuthUser(request.body as IUser);
             //Try authentication
-            const token = await AuthService.performAuthentication(parsedBody);
+            const token = await AuthService.performAuthentication(request.body);
             //Set token
             response.setHeader("Set-Cookie", "token=${token}; HttpOnly; Path=/; Max-Age=3600");
+            console.log("BIEN");
```
Comment (PRTTMPRPHT) [inline next-app/pages/api/auth/index.ts:27]: remove

File: next-app/src/components/presenter/login/login.presenter.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            })
+            .catch((error) => {
+                //Return error
+                console.error("Error:", error);
+                const errorString = error.toString();
+                setErrorString(errorString);
+            });
+    };
+
+    const validateForm = (data: LoginFormFields) => {
```
Comment (PRTTMPRPHT) [inline next-app/src/components/presenter/login/login.presenter.tsx:40]: Maybe not the best naming for sth that has the side-effect of logging in. nitpick, i know

File: next-app/src/components/presenter/login/login.presenter.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+const LoginPresenter: React.FC<LoginPresenterProps> = props => {
+
+    const router = useRouter();
+    const [errorString, setErrorString] = useState<string>("");
+
+    const loginUser = async (user: Partial<IUser>) => {
+        //Call register API
+        await fetch("/api/auth", {method: "POST", headers: {"Content-Type": "application/json",}, body: JSON.stringify(user),})
+            .then(async (response) => {
```
Comment (PRTTMPRPHT) [inline next-app/src/components/presenter/login/login.presenter.tsx:24]: no need to chain .then and .catch when using async/await

File: next-app/src/components/presenter/login/login.presenter.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import LoginView from "@/components/view/login/login.view";
+import { UserValidation } from "@/models/shared/user/user.validation";
+import { IUser } from "@/models/shared/user/user.interface";
+import { useRouter } from "next/navigation";
+
+interface LoginPresenterProps {}
+
+export interface LoginFormFields{
+    username: string,
+    password: string,
```
Comment (PRTTMPRPHT) [inline next-app/src/components/presenter/login/login.presenter.tsx:13]: with interfaces you terminate lines by semicolons

File: next-app/src/components/view/login/login.view.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+) => {
+
+    const [errorVisibility, setErrorVisibility] = useState<Boolean>(false);
+
+    useEffect(()=>{
+        if(errorString.length > 0)
+            setErrorVisibility(true);
+        else
+            setErrorVisibility(false);
+    }, [errorString]);
```
Comment (PRTTMPRPHT) [inline next-app/src/components/view/login/login.view.tsx:24]: no need for this? below you already do the conditional for errorVisibility, just replace that with `errorString.length > 0 ? "" : "hidden"`

File: next-app/src/components/view/login/login.view.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    const [formData, setFormData] = useState<LoginFormFields>({
+        username: "",
+        password: "",
+      });
```
[the remark this answers is not in the document]
Comment (PRTTMPRPHT) [reply next-app/src/components/view/login/login.view.tsx:29]: Oh what the hell, I think this is fine. If need be, I have a source we can show that does it similarly
https://kxie0124.medium.com/design-react-native-app-with-mvp-model-view-presenter-architecture-d451d35d6cac

File: next-app/src/components/view/register/register.view.tsx [hunk 7]
```diff
@@ -14,13 +14,22 @@ const RegisterView: React.FC<RegisterViewProps> = (
...
 ) =>{
 
+    const [errorVisibility, setErrorVisibility] = useState<Boolean>(false);
+
+    useEffect(()=>{
+        if(errorString.length > 0)
+            setErrorVisibility(true);
+        else
+            setErrorVisibility(false);
+    }, [errorString]);
```
Comment (PRTTMPRPHT) [inline next-app/src/components/view/register/register.view.tsx:24]: again no need for this

**Review by PRTTMPRPHT — APPROVED** [review]
overall, LGTM. few items you can tackle, nothing to block this over

File: next-app/src/components/presenter/login/login.presenter.tsx (hunk 3 above)
Comment (PRTTMPRPHT) [reply next-app/src/components/presenter/login/login.presenter.tsx]: this comment still applies

Comment (PRTTMPRPHT) [reply next-app/src/components/presenter/login/login.presenter.tsx]: nevermind github showed me a bad diff

File: next-app/src/components/presenter/register/register.presenter.tsx [hunk 8]
```diff
@@ -20,24 +20,43 @@ export interface RegisterFormFields{
 const RegisterPresenter: React.FC<RegisterPresenterProps> = props => {
     const router = useRouter();
     const [errorString, setErrorString] = useState<string>("");
+    const [ValidatingState, setValidatingState] = useState<Boolean>(false);
+    const [RegistrationCompleted, setRegistrationCompleted] = useState<Boolean>(false);
```
Comment (PRTTMPRPHT) [inline next-app/src/components/presenter/register/register.presenter.tsx:24]: this should be lowercase boolean

File: next-app/src/components/presenter/register/register.presenter.tsx [hunk 9]
```diff
@@ -20,24 +20,43 @@ export interface RegisterFormFields{
...
-                router.push("/");
+                if (response.status === 500)
+                    setErrorString("Internal Server Error");
+                else if (!response.ok) {
+                    const errorData = await response.json();
+                    setErrorString(errorData.error);
+                }else{
+                    setRegistrationCompleted(true);
+                }
+
```
Comment (PRTTMPRPHT) [inline next-app/src/components/presenter/register/register.presenter.tsx:45]: here, async/await applies

**Comment by PRTTMPRPHT** [comment]
I need to take a close look at the navbar stuff tmrw. overall, I'm not opposed to a merge here though, just go ahead

File: next-app/src/components/presenter/register/register.presenter.tsx (hunk 9 above)
Comment (ferran98campos) [reply next-app/src/components/presenter/register/register.presenter.tsx:45]: Could you ellaborate?

**Pull request merged**